modules/solid_mechanics/src/auxkernels/ShellLocalCoordinatesAux.C

```diff
@@ -0,0 +1,79 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ShellLocalCoordinatesAux.h"
+#include "libmesh/utility.h"
+#include "libmesh/string_to_enum.h"
+
+registerMooseObject("SolidMechanicsApp", ShellLocalCoordinatesAux);
+
+InputParameters
+ShellLocalCoordinatesAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription(
+      "This AuxKernel stores a specific component of a shell element's local coordinate "
+      "vector in an auxiliary variable.");
+  params.addParam<std::string>("base_name", "Mechanical property base name");
+  MooseEnum property("first_local_vector second_local_vector normal_local_vector");
+  params.addRequiredParam<MooseEnum>(
+      "property",
+      property,
+      "The local axis to output, calculated on the shell element. Valid options are: "
+      "first_local_vector, second_local_vector and normal_local_vector");
```

Comment:
Listing valid options here is redundant as the MooseEnum already provides this information.